onchain/rollups/test/foundry/inputs/InputBox.t.sol

```diff
@@ -22,60 +24,82 @@ contract InputBoxTest is Test {
+                CanonicalMachine.INPUT_MAX_SIZE
             )
         );
         _inputBox.addInput(app, new bytes(max + 1));
     }
 
-    function testAddInput(address app, bytes[] calldata inputs) public {
-        uint256 numInputs = inputs.length;
-        bytes32[] memory returnedValues = new bytes32[](numInputs);
+    function testAddInput(address app, bytes[] calldata payloads) public {
+        uint256 numPayloads = payloads.length;
+        bytes32[] memory returnedValues = new bytes32[](numPayloads);
         uint256 year2022 = 1641070800; // Unix Timestamp for 2022
 
-        // assume #bytes for each input is within bounds
-        for (uint256 i; i < numInputs; ++i) {
-            vm.assume(
-                inputs[i].length <= CanonicalMachine.INPUT_PAYLOAD_MAX_SIZE
-            );
+        // assume #bytes for each payload is within bounds
+        for (uint256 i; i < numPayloads; ++i) {
+            vm.assume(payloads[i].length <= _getMaxInputPayloadLength());
         }
 
         // adding inputs
-        for (uint256 i; i < numInputs; ++i) {
+        for (uint256 i; i < numPayloads; ++i) {
             // test for different block number and timestamp
             vm.roll(i);
             vm.warp(i + year2022);
 
             vm.expectEmit(true, true, false, true, address(_inputBox));
-            emit IInputBox.InputAdded(app, i, address(this), inputs[i]);
+            bytes memory input = EvmAdvanceEncoder.encode(
+                address(this),
+                i,
+                payloads[i]
+            );
+            emit IInputBox.InputAdded(app, i, input);
 
-            returnedValues[i] = _inputBox.addInput(app, inputs[i]);
+            returnedValues[i] = _inputBox.addInput(app, payloads[i]);
 
             assertEq(i + 1, _inputBox.getNumberOfInputs(app));
         }
 
         // testing added inputs
-        for (uint256 i; i < numInputs; ++i) {
-            bytes32 inputHash = LibInput.computeInputHash(
-                address(this),
-                i, // block.number
-                i + year2022, // block.timestamp
-                i, // index
-                inputs[i]
+        for (uint256 i; i < numPayloads; ++i) {
+            bytes32 inputHash = keccak256(
+                abi.encodeCall(
+                    Inputs.EvmAdvance,
+                    (
+                        address(this),
+                        i, // block.number
+                        i + year2022, // block.timestamp
+                        i, // inputBox.length
+                        payloads[i]
+                    )
+                )
             );
             // test if input hash is the same as in InputBox
             assertEq(inputHash, _inputBox.getInputHash(app, i));
             // test if input hash is the same as returned from calling addInput() function
             assertEq(inputHash, returnedValues[i]);
         }
     }
+
+    function _getMaxInputPayloadLength() internal pure returns (uint256) {
```

Comment:
Why not store this constant value in an immutable variable in this contract called `MAX_INPUT_PAYLOAD_LENGTH`, and use it instead? Then, you can move the code of function to the constructor of the test contract.

Reply:
I think it's cleaner as a function alone